shared/src/commonMain/kotlin/com.macaosoftware.sdui.app/marketplace/amadeus/auth/plugin/AuthRepository.kt

```diff
@@ -0,0 +1,32 @@
+package com.macaosoftware.sdui.app.marketplace.amadeus.auth.plugin
+
+import dev.gitlive.firebase.Firebase
+import dev.gitlive.firebase.auth.FirebaseAuth
+import dev.gitlive.firebase.auth.FirebaseUser
+import dev.gitlive.firebase.auth.auth
+
+class AuthRepository {
```

Comment:
I normally call this classes UseCase not Repository.
Also using try-catch and return null sounds like an old pattern. You you use MacaoResult